packages/governance/test/test-param-manager.js

```diff
@@ -0,0 +1,228 @@
+  };
+  const { getParams, updateRatio } = buildParamManager([ratioDescription]);
+  t.deepEqual(getParams()[ratioKey], ratioDescription);
+  updateRatio(makeRatio(701, brand, BASIS_POINTS));
+  t.deepEqual(getParams()[ratioKey].value, makeRatio(701, brand, BASIS_POINTS));
+
+  t.throws(
+    () => updateRatio(18.1),
+    {
+      message: 'Ratio 18.1 must be a record with 2 fields.',
+    },
+    'value should be a ratio',
+  );
+});
+
+test('params one brand', async t => {
+  const brandKey = 'Brand';
+  const { brand: roseBrand } = makeIssuerKit('roses', AssetKind.SET);
+  const { brand: thornBrand } = makeIssuerKit('thorns');
+  const brandDescription = {
+    name: brandKey,
+    value: roseBrand,
+    type: ParamType.BRAND,
+  };
+  const { getParams, updateBrand } = buildParamManager([brandDescription]);
+  t.deepEqual(getParams()[brandKey], brandDescription);
+  updateBrand(thornBrand);
+  t.deepEqual(getParams()[brandKey].value, thornBrand);
+
+  t.throws(
+    () => updateBrand(18.1),
+    {
+      message: 'value for "Brand" must be a brand, was 18.1',
+    },
+    'value should be a brand',
+  );
+});
+
+test('params one unknown', async t => {
+  const stuffKey = 'Stuff';
+  const { brand: stiltonBrand } = makeIssuerKit('stilton', AssetKind.SET);
+  const stuffDescription = {
+    name: stuffKey,
+    value: stiltonBrand,
+    type: ParamType.UNKNOWN,
+  };
+  const { getParams, updateStuff } = buildParamManager([stuffDescription]);
+  t.deepEqual(getParams()[stuffKey], stuffDescription);
+  updateStuff(18.1);
+  t.deepEqual(getParams()[stuffKey].value, 18.1);
+});
+
+test('params duplicate entry', async t => {
+  const stuffKey = 'Stuff';
+  const { brand: stiltonBrand } = makeIssuerKit('stilton', AssetKind.SET);
+  t.throws(
+    () =>
+      buildParamManager([
+        {
+          name: stuffKey,
+          value: 37n,
+          type: ParamType.NAT,
+        },
+        {
+          name: stuffKey,
+          value: stiltonBrand,
+          type: ParamType.UNKNOWN,
+        },
+      ]),
+    {
+      message: `each parameter name must be unique: "Stuff" duplicated`,
+    },
+  );
+});
+
+test('params unknown type', async t => {
+  const stuffKey = 'Stuff';
+  const stuffDescription = {
+    name: stuffKey,
+    value: 'It was the best of times, it was the worst of times',
+    type: 'quote',
+  };
+  // @ts-ignore  illegal value for testing
+  t.throws(() => buildParamManager([stuffDescription]), {
+    message: 'unknown type guard "quote"',
+  });
+});
+
+test('params multiple values', t => {
+  const stuffKey = 'Stuff';
+  const natKey = 'Nat';
+  const { brand: parmesanBrand } = makeIssuerKit('parmesan', AssetKind.SET);
+  const cheeseDescription = {
+    name: stuffKey,
+    value: parmesanBrand,
+    type: ParamType.UNKNOWN,
+  };
+  const piDescription = {
+    name: natKey,
+    value: 314159n,
+    type: ParamType.NAT,
+  };
+  const { getParams, updateNat: _updateNat, updateStuff } = buildParamManager([
```

Comment:
```suggestion
  const { getParams, updateStuff } = buildParamManager([
```

Seems to be unused. Is this intentional? If so, let's add a comment

Reply:
It's unused, but I wanted it to be visible that `updateNat` is defined. Perhaps I'll just use it instead of making a big deal of it.